control-plane/catalog/to-consul/syncer.go

```diff
@@ -173,91 +174,91 @@ func (s *ConsulSyncer) watchReapableServices(ctx context.Context) {
 	// because we have no tracked services in our maps yet.
 	<-s.initialSync
 
+	// Run immediately the first time, then wait for the retry period
+	waitCh := time.After(0)
+	waitBeforeRetry := s.SyncPeriod / 4
+
+	for {
+		select {
+		case <-waitCh:
+			s.deregisterRemovedServices(ctx)
+			waitCh = time.After(waitBeforeRetry)
+		case <-ctx.Done():
+			return
+		}
+	}
+}
+
```

Comment:
Very nit here, maybe we could do something similar to the code base:
```
for {
	select {
	case <-time.After(s.SyncPeriod / 4):
		s.deregisterRemovedServices(ctx)
	case <-ctx.Done():
		return
	}
}
```
for the benefit of saving some variables and lines unless I overlooked some pitfalls.